### PR TITLE
chore/deploy with helm

### PR DESCRIPTION
**Pull request by ivanstanev**
- [ ] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)
- [ ] Potential release notes have been inspected

### What this does

Currently we have no coverage of testing the deployment of snyk-monitor with Helm. This change addresses this. We currently choose to deploy only during KinD tests, and assume that we can deploy with Helm for every other platform if this test succeeds.

### Notes for the reviewer

Run with `npm run test:integration:kind:helm`

### More information

- [Jira ticket RUN-825](https://snyksec.atlassian.net/browse/RUN-825)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -136,13 +136,10 @@ jobs:
     <<: *default_machine_config
     steps:
       - checkout
+      - setup_node12
```
Comment (agatakrajewska) [inline .circleci/config.yml:139]: I'm just trying to understand how that config works 🙂So 'setup-node12' is a new step, where is it defined?

Comment (ivanstanev) [reply .circleci/config.yml:139]: Check the changes commit by commit, not by overall diff

Comment (ivanstanev) [reply .circleci/config.yml:139]: Actually my bad, it's not in this PR

It's something I introduced in a recent PR, here's how it looks!
https://github.com/snyk/kubernetes-monitor/blob/staging/.circleci/config.yml#L32-L44

Comment (agatakrajewska) [reply .circleci/config.yml:139]: amazing thanks! :-)

File: .circleci/config.yml [hunk 2]
```diff
@@ -184,10 +184,7 @@ jobs:
       - run:
           name: Integration tests
           command: |
-            docker login --username ${DOCKERHUB_USER} --password ${DOCKERHUB_PASSWORD} &&
-            export IMAGE_TAG=$([[ "$CIRCLE_BRANCH" == "staging" ]] && echo "staging-candidate" || echo "discardable") &&
-            export KUBERNETES_MONITOR_IMAGE_NAME_AND_TAG=snyk/kubernetes-monitor:${IMAGE_TAG}-${CIRCLE_SHA1} &&
-            docker pull ${KUBERNETES_MONITOR_IMAGE_NAME_AND_TAG} &&
+            export KUBERNETES_MONITOR_IMAGE_NAME_AND_TAG=$(./scripts/circleci-jobs/setup-integration-tests.py)
```
Comment (Shesekino) [inline .circleci/config.yml:187]: omg this is amazing

File: package.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
     "test:system": "tap test/system --timeout=600",
     "test:integration": "DEPLOYMENT_TYPE=YAML TEST_PLATFORM=kind CREATE_CLUSTER=true tap test/integration/kubernetes.test.ts --timeout=900",
     "test:integration:kind": "DEPLOYMENT_TYPE=YAML TEST_PLATFORM=kind CREATE_CLUSTER=true tap test/integration/kubernetes.test.ts --timeout=900",
+    "test:integration:kind:helm": "DEPLOYMENT_TYPE=Helm TEST_PLATFORM=kind CREATE_CLUSTER=true tap test/integration/kubernetes.test.ts --timeout=900",
```
Comment (Shesekino) [inline package.json:12]: how about adding 
```"test:integration:kind:yaml"```
which would be a copy-paste of 
test:integration:kind":
and at some point delete test:integration:kind AND test:integration?

Comment (ivanstanev) [reply package.json:12]: I thought about it but preferred to do it in a future PR, wanted to scope the changes only to Helm, but definitely the intention to use :yalm and remove the other two 🙏

File: test/setup/deployers/helm.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  console.log(`Deployed ${imageOptions.nameAndTag} with pull policy ${imageOptions.pullPolicy}`);
+}
+
+async function downloadHelm(): Promise<void> {
+  console.log(`Downloading Helm ${helmVersion}...`);
+  const os = platform();
+  await exec(
+    `curl https://get.helm.sh/helm-v${helmVersion}-${os}-amd64.tar.gz | tar xfzO - ${os}-amd64/helm > ${helmPath}`,
+  );
+  chmodSync(helmPath, 0o755); // rwxr-xr-x
```
[the remark this answers is not in the document]
Comment (ivanstanev) [reply test/setup/deployers/helm.ts:43]: It shouldn't be:
```sh
-rwxr-xr-x    1 ivan  staff    40M 31 Mar 15:37 helm*
```

File: test/setup/deployers/helm.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { exec } from 'child-process-promise';
+import { platform } from 'os';
+
+import { IDeployer, IImageOptions } from './types';
+import { existsSync, chmodSync } from 'fs';
+
+const helmVersion = '3.0.0';
```
[the remark this answers is not in the document]
Comment (ivanstanev) [reply test/setup/deployers/helm.ts:7]: While it makes sense to simplify the repeating of "helm", I think it gets a bit harder to read further down in the code

File: test/setup/deployers/helm.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Shesekino) [reply test/setup/deployers/helm.ts:43]: but isn't 755 is rwx r-x r-x ?

[the remark this answers is not in the document]
Comment (Shesekino) [reply test/setup/deployers/helm.ts:43]: 🤦‍♂ that's exactly that, I just can't read

**Pull request merged**